### PR TITLE
Updated model form declarations

### PR DESCRIPTION
**Pull request by ptrelease**

### CODE REVIEW

File: source/documentation/data_dictionary/tables/_declarations.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+##declarations
+
+Records a declaration as JSONB, so it is searchable.
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_declarations.md:3]: I've been thinking about this. Do we need the declaration to be searchable? It's something that we'll be declaring ourselves - there'll never be dynamic content in there. As a result, I don't think we'd ever have a need to search it. I think the other type of JSON column might be more applicable in this case.

File: source/documentation/data_dictionary/tables/_application_declarations.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## application_declaration
+
+Records the links between an applications and its declarations.
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_application_declarations.md:3]:
```suggestion
Records the links between an application and its declarations.
```

File: source/documentation/data_dictionary/tables/_application_declarations.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+Records the links between an applications and its declarations.
+In some cases the declarations can be linked to an application 
+using the implicit 'person' links.  However, this becomes
+complicated if the User spans applications.
+
+```
+  id: UUID <<PK>>
+  application_id: UUID <<FK>>
+  declaration_id: UUID <<FK>>
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_application_declarations.md:11]:
```suggestion
  declaration_id: UUID <<FK>>
  created_at: Timestamp
  updated_at: Timestamp
```

File: source/documentation/data_dictionary/tables/_declarations.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+Records a declaration as JSONB, so it is searchable.  
+When a declaration is revised, or a new declaration added, 
+then a new row will be added to this table.
+
+```
+  id: UUID <<PK>>
+  declaration_type: String      The type of declaration
+  json: JSONB                   A structured version of a declaration
+  version: Integer              The version of the declaration
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_declarations.md:11]:
```suggestion
  version: Integer              The version of the declaration
  created_at: Timestamp
  updated_at: Timestamp
```

File: source/documentation/data_dictionary/tables/_person_application.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+##person_application
+
+Links an application to a person.  Also records the purpose of that
+relationship, such as legal signatory.  
+
+```
+  id: Integer <<PK>>
+  person_id: UUID <<FK>>        ID on the person table.
+  application_id: UUID <<FK>>   ID on the application_id.
+  relationship: Integer <<FK>>  Link to the relationship table.
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_person_application.md:10]:
```suggestion
  relationship: Integer <<FK>>  Link to the relationship table.
  created_at: Timestamp
  updated_at: Timestamp
```

File: source/documentation/data_dictionary/tables/_person_declaration.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+## person_declaration
+
+Records the relationship between a person and their declarations, 
+such as agreeing to user research.
+
+```
+  id: UUID <<FK>>
+  person_id: UUID <<FK>>
+  declaration_id: UUID <<FK>>
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_person_declaration.md:9]:
```suggestion
  declaration_id: UUID <<FK>>
  created_at: Timestamp
  updated_at: Timestamp
```

File: source/documentation/data_dictionary/tables/_relationship_type.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
+## relationship type
+
+Records the different kinds of relationship that might 
+exist between an application and person. For instance, there
+may be a person on an application that is a legal signatory.
+
+```
+  id: integer <<PK>>
+  name: String          Contains the relationship name
```
Comment (stuartmccoll) [inline source/documentation/data_dictionary/tables/_relationship_type.md:9]:
```suggestion
  name: String          Contains the relationship name
  created_at: Timestamp
  updated_at: Timestamp
```

File: source/documentation/design/data_model.html.md.erb [hunk 8]
```diff
@@ -220,11 +225,36 @@ classDiagram
...
       name: String
       created_at: Timestamp
       updated_at: Timestamp
     }
-
+    class person_application{
+      id: Integer **PK**
+      person_id: UUID **FK**
+      application_id: UUID **FK**
+      relationship: Integer **FK*
```
Comment (stuartmccoll) [inline source/documentation/design/data_model.html.md.erb:238]:
```suggestion
      relationship: Integer **FK*
      created_at: Timestamp
      updated_at: Timestamp
```

File: source/documentation/design/data_model.html.md.erb [hunk 9]
```diff
@@ -220,11 +225,36 @@ classDiagram
...
+    }
+    class person_declaration{
+      id: UUID **PK**
+      person_id: UUID **FK**
+      declaration_id: UUID **FK*
```
Comment (stuartmccoll) [inline source/documentation/design/data_model.html.md.erb:243]:
```suggestion
      declaration_id: UUID **FK*
      created_at: Timestamp
      updated_at: Timestamp
```

File: source/documentation/design/data_model.html.md.erb [hunk 10]
```diff
@@ -220,11 +225,36 @@ classDiagram
...
+     }
+    class relationship_type{
+      id: integer **PK**
+      name: String
```
Comment (stuartmccoll) [inline source/documentation/design/data_model.html.md.erb:247]:
```suggestion
      name: String
      created_at: Timestamp
      updated_at: Timestamp
```

File: source/documentation/design/data_model.html.md.erb [hunk 11]
```diff
@@ -220,11 +225,36 @@ classDiagram
...
+    }
+    class declarations{
+      id: UUID **PK**
+      declaration_type: String
+      json: JSONB
+      version: integer
```
Comment (stuartmccoll) [inline source/documentation/design/data_model.html.md.erb:253]:
```suggestion
      version: integer
      created_at: Timestamp
      updated_at: Timestamp
```

File: source/documentation/design/data_model.html.md.erb [hunk 12]
```diff
@@ -220,11 +225,36 @@ classDiagram
...
+    }
+    class application_declarations{
+      id: UUID **PK**
+      application_id: UUID **FK**
+      declaration_id: UUID **FK**
```
Comment (stuartmccoll) [inline source/documentation/design/data_model.html.md.erb:258]:
```suggestion
      declaration_id: UUID **FK**
      created_at: Timestamp
      updated_at: Timestamp
```

File: source/documentation/data_dictionary/tables/_declarations.md (hunk 1 above)
Comment (ptrelease) [reply source/documentation/data_dictionary/tables/_declarations.md:3]: Agreed.  Confirmed that a normal JSON column would be adequate, as unlikely to require the kind of indexing JSONB allows.

File: source/documentation/data_dictionary/tables/_person_application.md [hunk 13]
```diff
@@ -0,0 +1,11 @@
+##person_application
```
[the remark this answers is not in the document]
Comment (ptrelease) [reply source/documentation/data_dictionary/tables/_person_application.md:1]: No, but i'll add it for consistency as other changes are being made.

File: source/documentation/data_dictionary/tables/_declarations.md [hunk 14]
```diff
@@ -0,0 +1,12 @@
+##declarations
```
[the remark this answers is not in the document]
Comment (ptrelease) [reply source/documentation/data_dictionary/tables/_declarations.md:1]: No, but i'll add it for consistency as other changes are being made.

**Pull request merged**